code/obj/artifacts/artifact_items/dimensional_key.dm

```diff
@@ -0,0 +1,402 @@
+		for (var/turf/T as anything in fissure_entr.reachable_turfs)
+			if (!T.reachable_turfs)
+				T.reachable_turfs = list()
+			T.reachable_turfs += fissure_entr
+		var/area/artifact_fissure/A = get_area(inner_door)
+		A.update_visual_mirrors(entrance, entrance_dir)
+		new /obj/art_fissure_mirror_updater_dummy(fissure_entr, entrance, entrance_dir)
+		entrance.icon = 'icons/turf/floors.dmi'
+		entrance.icon_state = "darkvoid"
+		entrance.opacity = TRUE
+		entrance.name = "Thick void mist"
+		entrance.desc = "Void mist thick enough that you can't see through it.. How did this get here?"
+		RL_UPDATE_LIGHT(entrance)
+
+/datum/artifact/dimensional_key
+	associated_object = /obj/item/artifact/dimensional_key
+	type_name = "Dimensional key"
+	type_size = ARTIFACT_SIZE_TINY
+	rarity_weight = 200
+	validtypes = list("eldritch", "precursor")
+	react_xray = list(73, 90, 38, 4, "ANOMALOUS")
+	examine_hint = "It kinda looks like it's supposed to be inserted into something."
+
+	effect_activate(obj/O)
+		. = ..()
+		if (.)
+			return TRUE
+		var/obj/item/artifact/dimensional_key/artkey = O
+		var/datum/mapPrefab/allocated/allocated = get_singleton(/datum/mapPrefab/allocated/artifact_fissure)
+		artkey.fissure_region = allocated.load()
+		var/turf/T = artkey.fissure_region.get_center()
+		var/area/artifact_fissure/fissure_area = get_area(T)
+		fissure_area.fissure_region = artkey.fissure_region
+
+		mirrored_physical_zone_created = TRUE
+
+/****** Supporting items/atoms/etc. *******/
+
+ABSTRACT_TYPE(/obj/art_fissure_cross_dummy)
+/obj/art_fissure_cross_dummy
+	name = ""
+	desc = ""
+	invisibility = INVIS_ALWAYS
+	anchored = ANCHORED_ALWAYS
+	var/turf/exit_turf
+	var/required_dir
+
+	New(newLoc, turf/exit)
+		..()
+		src.exit_turf = exit
+
+	disposing()
+		src.exit_turf = null
+		..()
+
+	Crossed(atom/movable/AM)
+		if (istype(AM, /obj/projectile))
+			var/obj/projectile/P = AM
+			var/obj/projectile/new_proj = initialize_projectile(src.exit_turf, P.proj_data, P.xo, P.yo, P.shooter)
+			new_proj.travelled = P.travelled
+			new_proj.launch()
+			P.die()
+		else if (AM.dir == src.required_dir && !istype(AM, /obj/artifact_fissure_door) && !istype(AM, /obj/art_fissure_cross_dummy))
+			AM.set_loc(src.exit_turf)
+		else
+			return ..()
+
+	north
+		required_dir = NORTH
+
+	east
+		required_dir = EAST
+
+	south
+		required_dir = SOUTH
+
+	west
+		required_dir = WEST
+
+/obj/art_fissure_mirror_updater_dummy
+	name = ""
+	desc = ""
+	invisibility = INVIS_ALWAYS
+	anchored = ANCHORED_ALWAYS
+	var/turf/station_ref
+	var/entrance_loc
+
+	New(newLoc, turf/station_ref, entrance_loc)
+		..()
+		src.station_ref = station_ref
+		src.entrance_loc = entrance_loc
+
+	disposing()
+		src.station_ref = null
+		..()
+
+	Crossed(atom/movable/AM)
+		if (isliving(AM) && !isintangible(AM))
+			var/area/artifact_fissure/fissure_area = get_area(src)
+			fissure_area.update_visual_mirrors(src.station_ref, src.entrance_loc)
+			return ..()
+		return ..()
+
+/obj/artifact_fissure_door
```

Comment:
i feel like we should put this under something, even if its abstract or similar, just for organization

Reply:
I think I did what you were thinking, let me know if not